### PR TITLE
Updated standard library docs.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Cleaned up and re-worded some standard library documentation.

Co-authored-by: Devon Campbell <devon@edgedb.com>

Co-authored-by: Sophia <41456914+i0bs@users.noreply.github.com>

Co-authored-by: Yury Selivanov <yury@edgedb.com>

### CODE REVIEW

**Comment by vpetrovykh** [comment]
This is a cleaned up version of #4464

File: docs/stdlib/datetime.rst [hunk 1]
```diff
@@ -498,30 +500,31 @@ EdgeDB stores and outputs timezone-aware values in UTC.
...
         select <cal::date_duration>'3 weeks 5 days';
         select <cal::date_duration>'-7 millennia';
 
-    Generally ``date_duration`` is fully compatible with
-    :eql:type:`cal::relative_duration` and has the same general behaviour and
-    caveats. It will even to implicitly cast to
-    :eql:type:`cal::relative_duration` in any situation where only
-    :eql:type:`cal::relative_duration` is expected.
+    In most cases, ``date_duration`` is fully compatible with
+    :eql:type:`cal::relative_duration` and shares the same general behaviour
```
Comment (raddevon) [inline docs/stdlib/datetime.rst:520]:
```suggestion
    :eql:type:`cal::relative_duration` and shares the same general behavior
```

File: docs/stdlib/numbers.rst [hunk 2]
```diff
@@ -789,14 +817,11 @@ from :eql:type:`str` and :eql:type:`json`.
...
     :index: parse int16
 
-    Create an :eql:type:`int16` value.
-
-    Parse an :eql:type:`int16` from the input *s* and optional format
-    specification *fmt*.
-
-    For more details on formatting see :ref:`here
-    <ref_std_converters_number_fmt>`.
+    Returns a :eql:type:`int16` value parsed from the given string.
```
Comment (raddevon) [inline docs/stdlib/numbers.rst:820]:
```suggestion
    Returns an :eql:type:`int16` value parsed from the given string.
```

File: docs/stdlib/numbers.rst [hunk 3]
```diff
@@ -805,13 +830,11 @@ from :eql:type:`str` and :eql:type:`json`.
 
     :index: parse int32
 
-    Create an :eql:type:`int32` value.
-
-    Parse an :eql:type:`int32` from the input *s* and optional format
-    specification *fmt*.
+    Returns a :eql:type:`int32` value parsed from the given string.
```
Comment (raddevon) [inline docs/stdlib/numbers.rst:833]:
```suggestion
    Returns an :eql:type:`int32` value parsed from the given string.
```

File: docs/stdlib/numbers.rst [hunk 4]
```diff
@@ -821,13 +844,11 @@ from :eql:type:`str` and :eql:type:`json`.
 
     :index: parse int64
 
-    Create an :eql:type:`int64` value.
+    Returns a :eql:type:`int64` value parsed from the given string.
```
Comment (raddevon) [inline docs/stdlib/numbers.rst:847]:
```suggestion
    Returns an :eql:type:`int64` value parsed from the given string.
```

File: docs/stdlib/sequence.rst [hunk 5]
```diff
@@ -66,12 +70,17 @@ Sequences
...
+    See the note on :ref:`specifying your sequence
+    <ref_std_specifying_sequence>` for best practices on supplying the *seq*
+    parameter.
 
     The single-parameter form resets the sequence to its initial state, where
     the next :eql:func:`sequence_next` call will return the first value in
-    sequence.  The two-parameters form allows changing the last returned
-    value of the sequence.
+    sequence.  The two-parameter form allows you to set the current value of
+    the sequence. The next :eql:func:`sequence_next` call will return the value
```
Comment (raddevon) [inline docs/stdlib/sequence.rst:82]:
```suggestion
    sequence. The two-parameter form allows you to set the current value of the
    sequence. The next :eql:func:`sequence_next` call will return the value
```

**Review by raddevon — APPROVED** [review]
Fantastic, @vpetrovykh! Left a few very minor suggestions.

**Comment by msullivan** [comment]
Is this done? Should we merge it?

**Pull request merged**